src/tcp.cpp

```diff
@@ -0,0 +1,189 @@
+#include "tcp.hpp"
+
+int TCP::EstablishServerSocket()
+{
+	// server address config
+	struct sockaddr_in address;
+	address.sin_family = AF_INET;
+	address.sin_addr.s_addr = INADDR_ANY;
+	address.sin_port = htons(atoi(this->port));
+	// AF_INET = ipv4, SOCK_STREAM = tcp, 0 = ip
+	int serverSocket = socket(AF_INET, SOCK_STREAM, 0);
+	if (serverSocket == -1) {
+		perror("socket failure");
+		close(serverSocket);
+		return -1;
+	}
+
+	// connects the socket to the address
+	if (bind(serverSocket, (struct sockaddr*)&address, sizeof(address)) == -1) {
+		perror("bind failure");
+		close(serverSocket);
+		return -1;
+	}
+	return serverSocket;
+}
+
+void TCP::addConnection(int connectionSocket)
+{
+    //adds another spot to bitfield vector
```

Comment:
Connections have nothing to do with the bitfield. The bitfield is a map that keeps track of whether the peer has a certain piece of the file.

Imagine a file of size 5KB (file_size), with a piece size of 1KB (piece_size). Then, we have a total of file_size/piece_size pieces. In this case, there are 5 pieces.

The bitfield is thus 5 bits long, and would look something like this: [ 0 0 0 0 0 ]

If we have the first piece in the file (the first 1KB), you set the zero index to true (i.e. 1): [ 1 0 0 0 0 ]
If you have the second piece, you set the first index to 1: [ 0 1 0 0 0 ]

The bitfield just tracks which pieces you have, not your active connections

Reply:
When a new connection is added we have to store its bitfield so we can request specific pieces we are missing, this line says that it adds another index to the vector that stores the bitfields in preparation for when we handshake and then the peer's bitfield is passed and we have to then put it in.

Incrementing the index is done here because it avoids threading issues because if the handler is spin up is slow or communication to the other connection is slow it the indices could misalign and a bitfield could be matched with a different connection, I mentioned that a struct would be a good solution to this in the comments in the header file.

I believe that it is useful to store the bitfields of the other connections so we can iterate through them and make decisions on who to request pieces from.

It is not the best implementation of the idea but this is only a midpoint review and this should work as long as there are no errors, I plan to implement the struct later.